app/start_from_snapshot.go

```diff
@@ -0,0 +1,198 @@
+package app
+
+import (
+	"archive/tar"
+	"bytes"
+	"fmt"
+	"github.com/klauspost/pgzip"
+	"github.com/okx/okbchain/libs/cosmos-sdk/types/errors"
+	"github.com/okx/okbchain/libs/tendermint/libs/log"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+)
+
+func prepareSnapshotDataIfNeed(snapshotURL string, home string, logger log.Logger) {
+	if snapshotURL == "" {
+		return
+	}
+
+	snapshotHome := filepath.Join(home, ".download_snapshots")
+
+	// check whether the snapshot file has been downloaded
+	byteData, err := os.ReadFile(filepath.Join(snapshotHome, ".record"))
+	if err == nil && strings.Contains(string(byteData), snapshotURL) {
+		return
+	}
+
+	if _, err := url.Parse(snapshotURL); err != nil {
+		panic(errors.Wrap(err, "invalid snapshot URL"))
+	}
+
+	// download snapshot
+	snapshotFile, err := downloadSnapshot(snapshotURL, snapshotHome, logger)
+	if err != nil {
+		panic(err)
+	}
+
+	// uncompress snapshot
+	logger.Info("start to uncompress snapshot")
+	if err := extractTarGz(snapshotFile, snapshotHome); err != nil {
+		panic(err)
+	}
+
+	// delete damaged data
+	logger.Info("start to delete damaged data")
+	if err := os.RemoveAll(filepath.Join(home, "data")); err != nil {
+		panic(err)
+	}
+
+	// move snapshot data
+	logger.Info("start to move snapshot data")
+	if err := moveDir(filepath.Join(snapshotHome, "data"), filepath.Join(home, "data")); err != nil {
+		panic(err)
+	}
+
+	os.Remove(snapshotFile)
+
+	os.WriteFile(filepath.Join(snapshotHome, ".record"), []byte(snapshotURL+"\n"), 0644)
+
+	logger.Info("snapshot data is ready, start node soon!")
+}
+
+func downloadSnapshot(url, outputPath string, logger log.Logger) (string, error) {
+	// create dir
+	_, err := os.Stat(outputPath)
+	if err != nil {
+		os.MkdirAll(outputPath, 0755)
+	}
+
+	fileName := url[strings.LastIndex(url, "/")+1:]
+	targetFile := filepath.Join(outputPath, fileName)
+
+	// check file exists
+	if _, err := os.Open(targetFile); err == nil {
+		os.Remove(targetFile)
+	}
+
+	//maxSpeed := fmt.Sprintf("%d", viper.GetInt(server.FlagMaxDownloadSnapshotSpeed)*1024*1024)
+	var stdoutProcessStatus bytes.Buffer
+
+	axel := exec.Command("axel", "-n", fmt.Sprintf("%d", runtime.NumCPU()), "-o", targetFile, url)
```

Comment:
`axel` is our provided tools for fast download? if not installed, what will happen?

Reply:
it got installed in the base image